packages/expo-ads-facebook/src/withNativeAd.tsx

```diff
@@ -78,18 +84,27 @@ export default function withNativeAd<P>(
       let { adsManager } = this.props;
       let props = this._getForwardedProps();
       return (
-        <NativeAdView
-          ref={this._nativeAdViewRef}
-          adsManager={adsManager.placementId}
-          onAdLoaded={this._handleAdLoaded}>
-          <AdMediaViewContext.Provider value={this._adMediaViewContextValue}>
-            <AdIconViewContext.Provider value={this._adIconViewContextValue}>
-              <AdTriggerViewContext.Provider value={this._adTriggerViewContextValue}>
-                {this.state.ad ? <Component {...props} nativeAd={this.state.ad} /> : null}
-              </AdTriggerViewContext.Provider>
-            </AdIconViewContext.Provider>
-          </AdMediaViewContext.Provider>
-        </NativeAdView>
+        <NativeAdLayout>
```

Comment:
Let's change it to `const viewHierarchy = (<NativeAdView>…)`

and `if (android) { return <NativeAdLayout>{viewHierarchy}} else viewHierarchy`